elftools/elf/notes.py

```diff
@@ -49,7 +49,8 @@ def iter_notes(elffile, offset, size):
         elif note['n_type'] == 'NT_GNU_PROPERTY_TYPE_0':
             off = offset
             props = []
-            while off < end:
+            current_note_end = offset + note['n_descsz']
```

Comment:
Can you add a small comment here explaining the semantics (the usage of `n_descsz`)? A link to some spec-like thing would be great too, if possible